app/src/components/mode-specific/desktop/InterceptTraffic/WebTraffic/TrafficTableV2/utils/index.ts

```diff
@@ -48,3 +53,147 @@ export const doesStatusCodeMatchLabels = (code: number = 0, labels: STATUS_CODE_
+    responseRulePair.source.key = GLOBAL_CONSTANTS.RULE_KEYS.URL;
+    responseRulePair.source.operator = GLOBAL_CONSTANTS.RULE_OPERATORS.CONTAINS;
+  }
+
+  let operationParams = null;
+
+  if (ruleParams.resourceType === "graphqlApi" && ruleParams.operationKeys.length) {
+    const sourceFilters = responseRulePair.source.filters[0] ?? {};
+    operationParams = getGraphQLOperationValues(JSON.parse(ruleParams.requestDetails.body), ruleParams.operationKeys);
+    sourceFilters.requestPayload = {};
+
+    if (operationParams) {
+      sourceFilters.requestPayload.key = operationParams?.operationKey;
+      sourceFilters.requestPayload.value =
+        typeof operationParams?.operationValue === "string"
+          ? operationParams?.operationValue
+          : JSON.stringify(operationParams?.operationValue);
+      responseRulePair.source.filters = [sourceFilters];
+    }
+  }
+
+  return {
+    ...newResponseRule,
+    name: `[MOCK] ${operationParams?.operationValue || ""}-${ruleParams.requestUrl}`,
+    description: `Mock response for ${ruleParams.requestUrl} - ${operationParams?.operationValue || ""}`,
+    groupId: ruleParams.groupId,
+    sessionCreated: true,
+    pairs: [responseRulePair],
+    status: GLOBAL_CONSTANTS.RULE_STATUS.ACTIVE,
+  };
+};
+
+export const getGraphQLOperationValues = (
+  requestData: any,
+  operationKeys: string[]
+): {
+  operationKey: string;
+  operationValue: string;
+} | null => {
+  // We only allow request payload targeting when requestData is JSON
+  if (!requestData || typeof requestData !== "object") return null;
+  if (Object.keys(requestData).length === 0) return null;
+
+  let operationValueInRequestData: string | null = null;
+  let operationKeyInRequestData: string | null = null;
+
+  // tagettedKey is the json path e.g. a.b.0.c
+  if (operationKeys.length) {
+    operationKeys.some((key) => {
+      const valueInRequestData = traverseJsonByPath(requestData, key);
+      if (valueInRequestData) {
+        operationKeyInRequestData = key;
+        operationValueInRequestData = valueInRequestData;
+        return true;
+      }
+      return false;
+    });
+  }
+
+  if (operationValueInRequestData) {
+    return { operationKey: operationKeyInRequestData, operationValue: operationValueInRequestData };
+  }
+
+  return null;
+};
+
+/**
+ * @param {Object} json
+ * @param {String} path -> "a", "a.b", "a.0.b (If a is an array containing list of objects"
+ * Also copied in shared/utils.js for the sake of testing
+ */
+export const traverseJsonByPath = (jsonObject: any, path: any) => {
+  if (!path) return;
+
+  const pathParts = path.split(".");
+
+  try {
+    let i = 0;
+    // Reach the last node but not the leaf node.
+    for (i = 0; i < pathParts.length - 1; i++) {
+      jsonObject = jsonObject[pathParts[i]];
+    }
+
+    return jsonObject[pathParts[pathParts.length - 1]];
+  } catch (e) {
+    /* Do nothing */
+    console.log(e);
+  }
+};
```

Comment:
This is already exported from requestly/common
Can we use it from there?

Reply:
That is not exported from common/utils. Logged a task for this